relayer/processor/path_end_runtime.go

```diff
@@ -617,6 +615,7 @@ func (pathEnd *pathEndRuntime) removePacketRetention(
 		toDelete[preInitKey] = []uint64{sequence}
 	case chantypes.EventTypeRecvPacket:
 		toDelete[eventType] = []uint64{sequence}
+		toDelete[chantypes.EventTypeWriteAck] = []uint64{sequence}
```

Comment:
why do we try to delete `EventTypeWriteAck` on the recv also?

Reply:
if we give up on a recv, we may also have the associated write acknowledgement event cached. Without this, it would leak because it would never be cleaned up.